cmd/kubeadm/app/phases/uploadconfig/uploadconfig.go

```diff
@@ -32,121 +37,159 @@ import (
 	rbachelper "k8s.io/kubernetes/pkg/apis/rbac/v1"
 )
 
+type updateKubeadmConfigMap func() error
+
 const (
 	// NodesKubeadmConfigClusterRoleName sets the name for the ClusterRole that allows
 	// the bootstrap tokens to access the kubeadm-config ConfigMap during the node bootstrap/discovery
 	// or during upgrade nodes
 	NodesKubeadmConfigClusterRoleName = "kubeadm:nodes-kubeadm-config"
 )
 
+var (
+	lockIdentity = string(uuid.NewUUID())
+)
+
 // ResetClusterStatusForNode removes the APIEndpoint of a given control-plane node
 // from the ClusterStatus and updates the kubeadm ConfigMap
-func ResetClusterStatusForNode(nodeName string, client clientset.Interface) error {
+func ResetClusterStatusForNode(cfg *kubeadmapi.InitConfiguration, client clientset.Interface) error {
```

Comment:
we're going to need an e2e test or partial integration test using multiple writers.